.env.sample

```diff
@@ -115,4 +115,7 @@ RUN_AT_TIMES =
 # An example to match the above would be 0 2,4 * * *
 # This should run as frequently or more frequently than RUN_AT_TIMES
 # Use https://crontab.guru to validate your schedule
-CRONTAB_SCHEDULE = 
\ No newline at end of file
+CRONTAB_SCHEDULE = 
+
+# Change this to set the max default weeks to allow. Default is currently 16. The issue is the end dates in Canvas currently are set 10 years out so it can't calculate the range.
```

Comment:
What is the need for setting the max default week value? can we have calculate this based on the start and end date of the term and add it to the json feed as a property like
```
{
....
"max_week":8
...
}
```

Reply:
Yeah I am calculating it.

However some weeks in Canvas are set to many years in the future per KB0023342. I do have code in the view that will set the end_year to be the same as the start_year if the year difference is > 1. This should fix this canvas issue, and this really probably isn't ever needed.  It does this calculation correctly if the input is correct.

We talked about using a different source of data (like UM API) to pull the "actual" term dates (not the ones coded into Canvas) but this would be extra effort to do this as well as match them up to the canvas terms. Maybe if it's necessary in the future.

Reply:
Yes, I do feel the need to get the start and end term date from Some UM API. We can't trust canvas course start and end date as that is extended for some business reasons. 
 As for I know, we are still hard-coding the start and end term date in DB so I think we can safely assume that we get the start/end date no matter how it is fed to the DB table( via API or manual placing)

Reply:
After some recent changes it's actually pulling the academic term dates down from Canvas, we don't have local control over that anymore in our database. 
https://github.com/tl-its-umich-edu/my-learning-analytics/blob/b8f2659ca869b59d7b45d4d7d885e46fe4137ceb/dashboard/cron.py#L338

I thought we could at least trust the "dates" if not the years but it looks like the dates for winter are even incorrect

Winter 2019 = Term Runs from | Jan 9 at 12am | to | Dec 31, 2029 at 12am

I'll have to see what's the easiest fix for this. I'll check with Dan, maybe we can at least guarantee the "Date" is correct if not the year. This is the first time it looks like that wasn't the case.

Reply:
This was resolved, we set the dates correctly.

Reply:
For the assignments view there is no need for the end date but only start date. So based on the start date I am calculating the week and if you have a question how do I know how many weeks then I don't need to know when the term end, as assignment due, is what I care or student care. So in Assignment view there is not burning need. 

But it will useful if we have the term end date. 
I don't know pulling that start/end date from canvas is a good idea, but with DPDB we might get it from there